pkg/utils/kubeutils/portforward/cli_portforwarder.go

```diff
@@ -61,6 +61,8 @@ func (c *cliPortForwarder) startOnce(ctx context.Context) error {
 		cmdCtx,
 		"kubectl",
 		"port-forward",
+		"--context",
```

Comment:
Could you provide more context about why you wouldn't expect it to succeed? I don't think that line uses the `cliPortForwarder`, so I think it's unrelated? And it passes the kubecontext to the command as an arg. But maybe I'm missing how these are related.

Reply:
https://github.com/solo-io/gloo/blob/main/projects/gloo/cli/pkg/cmd/check/root.go#L247 when we check anything that requires talking to a specific pod (like the code) I linked,

https://github.com/solo-io/gloo/blob/main/projects/gloo/cli/pkg/cmd/check/proxy_prom_stats.go#L26

we open a port-forward using the "old" cliport-forwarder: https://github.com/solo-io/gloo/blob/main/projects/gloo/cli/pkg/cmd/check/proxy_prom_stats.go#L69

that calls the legacy code: https://github.com/solo-io/gloo/blob/main/pkg/cliutil/uri.go#L195, which as the comment explains, calls our preferred code.

In the earlier days we had a lot of duplicated port-forwarding code. The benift is that now all of it shares the same source functionality. The downside is that if there is a bug, it is more widespread (I would actually argue that's good). 

What I'm curious about is how we have `glooctl check` tests that claim to pass, but I'm surprised that they could without this change

Reply:
Oh 🤦‍♀️, I did not at all correctly process what the code you linked is doing.

The reason the negative case passes is because the first k8s interaction in `glooctl check` is a call to `[checkConnection](https://github.com/solo-io/gloo/blob/v1.19.0-beta1/projects/gloo/cli/pkg/cmd/check/root.go#L988)`, which does correctly use `opts.Top.KubeContext`.

My first guess is that the positive case passes because it's the default k8s context and the passed in context is irrelevant? Testing [here](https://github.com/solo-io/gloo/pull/10461).

Reply:
The test passed with the `--kube-context` flag omitted, which supports my theory. And I suspect this is far from the only place in our CLI where that flag is not fully respected. Perhaps it's worth an issue focused on that flag.

Reply:
https://github.com/solo-io/gloo/issues/10462